### ISSUE TITLE
perf: `trtllm_ragged_attention` requires zeroed output and workspace for non-causal attention

### ISSUE THREAD

**Issue by mgoin** [flashinfer-ai/flashinfer#2409]
## Summary

When using `trtllm_ragged_attention` (via `trtllm_ragged_attention_deepseek` in vLLM) for non-causal attention (`is_causal=False`) with variable sequence lengths, the output tensor and workspace buffer must be pre-zeroed by the caller for correct results. This adds significant overhead on the critical path during chunked prefill.

## Expected Behavior

The kernel should handle uninitialized output and workspace memory internally, producing correct results without requiring the caller to zero-fill tensors beforehand.

## Context

In vLLM's MLA attention implementation, we use `trtllm_ragged_attention_deepseek` for chunked context prefill (non-causal attention over cached KV). We discovered that accuracy dropped significantly without explicit zeroing:

```python
# vllm/model_executor/layers/attention/mla_attention.py

def _run_prefill_context_chunk_trtllm_ragged(
    self, prefill: MLACommonPrefillMetadata, chunk_idx: int, q, k, v
):
    """TRT-LLM ragged attention for context chunks (non-causal)."""
    from flashinfer.prefill import trtllm_ragged_attention_deepseek

    # ISSUE: These zero-fills are required for correctness but add overhead on the hot path
    out = torch.zeros(
        q.shape[0],
        q.shape[1],
        v.shape[2],
        device=q.device,
        dtype=q.dtype,
    )
    prefill.workspace_buffer.fill_(0)

    attn_out, lse = trtllm_ragged_attention_deepseek(
        query=q,
        key=k,
        value=v,
        workspace_buffer=prefill.workspace_buffer,
        seq_lens=prefill.chunked_context.seq_lens[chunk_idx],
        # ... other params ...
        is_causal=False,  # Non-causal for context chunks
        return_lse=True,
        out=out,
    )
```

NOTE: The causal case (`_run_prefill_new_tokens_trtllm_ragged`) does NOT require this zeroing and works correctly without it.

**Comment by pavanimajety** [comment]
@PerkzZheng for viz

**Comment by PerkzZheng** [comment]
@mgoin this kernel doesn't require zero initialisation of the output tensor. Can you prepare a standalone test so that we can reproduce this issue easily ?

**Comment by saltyminty** [comment]
@mgoin checking again if you're still running into this issue, and if so, do you have a standalone repro?
